pkg/config/config_test.go

```diff
@@ -0,0 +1,54 @@
+package config
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLoadConfig(t *testing.T) {
+	// Create a temporary directory for test files
+	tmpDir, err := os.MkdirTemp("", "config_test")
+	if err != nil {
+		t.Fatalf("Failed to create temp directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	// Create an empty file for testing
+	configFilePath := filepath.Join(tmpDir, ".sauced.yaml")
+
+	if err := os.WriteFile(configFilePath, []byte{}, 0644); err != nil {
+		t.Fatalf("Failed to create empty file: %v", err)
+	}
+
+	// Path for a non-existent file
+	nonExistentPath := filepath.Join(tmpDir, "non_existent.yaml")
```

Comment:
I'd move all of the setup above into the nested `t.Run(tt.name, func(t *testing.T) {`. That way, each test is completely independent and has all its own setup with its own temporary directory and temporary file.

Otherwise you run the risk of tests colliding or potentially impacting eachother if they were to mutate the file or dir in some way.

Reply:
Yeah makes sense. I'd normally do that. I found an example and was just happy to get it running. 😅